v2/features/farms/FarmsTableRowBody.tsx

```diff
@@ -10,11 +13,39 @@ import { CoreSelectors } from "v2/store/core";
 import { UserData } from "picklefinance-core/lib/client/UserModel";
 import { UserSelectors } from "v2/store/user";
 import { getUserAssetDataWithPrices } from "./FarmsTableRowHeader";
+import { ChainNetwork } from "picklefinance-core";
+import { Network } from "../connection/networks";
+import { switchChain } from "../connection/ConnectionStatus";
 
 interface Props {
   jar: JarDefinition;
 }
 
+const ConnectButton: FC<{ network: Network | undefined }> = ({ network }) => {
+  const { library } = useWeb3React<Web3Provider>();
+  const allCore = useSelector(CoreSelectors.selectCore);
+
+  if (!network || !allCore || !library) return null;
+  return (
+    <Button onClick={() => switchChain(library, network.chainId, allCore)}>
+      <span>Connect to</span>
+      <div className="w-4 h-4 mr-1 ml-1">
+        <Image
+          src={`/networks/${network.name}.png`}
+          width={200}
+          height={200}
+          layout="responsive"
```

Comment:
Please make this `intrinsic` for the smallest file size. That will make the image the size the wrapping div (it already has fixed width) and add 2x and 3x srcsets for higher DPI screens.

Reply:
done!